File: app/src/main/java/com/zlm/kt/connect/scheduler/BaseSchedulerProviderImp.java
package com.zlm.kt.connect.scheduler;


import androidx.annotation.NonNull;
import io.reactivex.ObservableTransformer;
import io.reactivex.Scheduler;

/**
 * @author Milla
 * @create 2020/3/27
 */
public interface BaseSchedulerProviderImp {

    @NonNull
    Scheduler computation();

    @NonNull
    Scheduler io();

    @NonNull
    Scheduler ui();

    @NonNull
    <T> ObservableTransformer<T, T> applySchedulers();
}

File: app/src/main/java/com/zlm/kt/connect/scheduler/BaseSchedulerProviderImp.kt
package com.zlm.kt.connect.scheduler

import io.reactivex.ObservableTransformer
import io.reactivex.Scheduler

/**
 * @author Milla
 * @create 2020/3/27
 */
interface BaseSchedulerProviderImp {
    fun computation(): Scheduler
    fun io(): Scheduler
    fun ui(): Scheduler
    fun <T> applySchedulers(): ObservableTransformer<T, T>
}